modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java

```diff
@@ -151,6 +153,10 @@ public String getHelp() {
 
     @Override
     public void processOpts() {
+        if (WEBCLIENT.equals(getLibrary()) && "threetenbp".equals(dateLibrary)) {
```

Comment:
I guess that this check is too early. I am not sure, but I think that if the value is set to something on the command line, the value will be set with `super.processOpts()` and will override your check.

If possible I would override the wrong/unsupported values in the "else if" block line 290.